### ISSUE TITLE
execpi interval values are not correct

### ISSUE THREAD

**Issue by plikhari** [brndnmtthws/conky#870]
This is from version 1.11.4 release - 1.11.3 works as designed.

A value for `execpi 3600` means every 3600 seconds the event is triggered.

In this case the trigger interval is 84 seconds and if I reduce the required value to 1800 then it drops to 42 seconds.

This does not affect conkywx as I use internal validation - but on the terminal window it shows up conky trying to update the program.

```
Conkywx program execution completed in: 0m0.982s on 31st at 07:53:00 pm
Conkywx program execution completed in: 0m1.115s on 31st at 07:54:13 pm
Conkywx program execution completed in: 0m0.846s on 31st at 07:55:37 pm
Conkywx program execution completed in: 0m0.911s on 31st at 07:57:01 pm
Conkywx program execution completed in: 0m0.909s on 31st at 07:58:25 pm
Conkywx program execution completed in: 0m1.021s on 31st at 07:59:49 pm
Conkywx program execution completed in: 0m0.995s on 31st at 08:01:13 pm
Conkywx program execution completed in: 0m0.900s on 31st at 08:02:37 pm
Conkywx program execution completed in: 0m0.897s on 31st at 08:04:01 pm
```

I understand git bisect will probably be required but I am strapped for time and can at present only highlight bugs...

cheers

**Comment by Tomunu** [comment]
You might already know this, but commands used with `execpi` are evaluated at every interval, even though the display doesn't change. But I don't know why it would get run twice as often after changing the `execpi` interval, even though that interval is still way higher than the Conky `update_interval`. So, that probably doesn't explain this problem (especially since the problem only started with version 1.11.4) but here are the descriptions of the `execp` and `execpi` objects anyway. 

execp command
Executes a shell command and displays the output in conky. warning: this takes a lot more resources than other variables. I'd recommend coding wanted behaviour in C and posting a patch. This differs from $exec in that it parses the output of the command, so you can insert things like ${color red}hi!${color} in your script and have it correctly parsed by Conky. Caveats: Conky parses and evaluates the output of $execp every time Conky loops, and then destroys all the objects. If you try to use anything like $execi within an $execp statement, it will functionally run at the same interval that the $execp statement runs, as it is created and destroyed at every interval.

execpi interval command
Same as execp but with specific interval. Interval can't be less than update_interval in configuration. Note that the output from the $execpi command is still parsed and evaluated at every interval.

**Comment by plikhari** [comment]
> You might already know this, but commands used with execpi are evaluated at every interval, even though the display doesn't change.

Yes 

>But I don't know why it would get run twice as often after changing the execpi interval, even though that interval is still way higher than the Conky update_interval

Precisely the reason I raised this flag. 
The update_interval is 3 seconds and the execpi is 3600 seconds - even though the two are related and that internally conky checks every 3 seconds - but it is happening every 84 seconds. Internally it is probably happening every (84 / 3600) * 3 value (0.07). I have a smooth scroller working at that update_interval of 0.07...

People will have an issue with this bug.

**Comment by Tomunu** [comment]
In that time period between 1.11.3 and 1.11.4, the only explicit changes to the `exec` objects were related to nesting those expressions. Do you have any nested `exec` objects?

**Comment by plikhari** [comment]
No - only a single line - and anything later only uses lua to execute other processes

```
conky.text = [[
${execpi 3600 ${template0} -c ${template1} -t ${template2} }
]];
```

**Comment by 7oxicshadow** [comment]
Was having issues with execi and execpi intervals being incorrect myself and came across this bug.

I realise this has been open a while now but for me the solution was to comment out/delete the 
`conky::run_all_callbacks();`
line from conky,cc (line740) [in the evaluate function].

Because this function is already being called elsewhere it is screwing the timings up having it being called more than once.

**Comment by plikhari** [comment]
@7oxicshadow 
Wow - thanks - shall test that out - I guess over time things will iron themselves out and we shall step over 1.11.3 version as well lol

**Comment by 7oxicshadow** [comment]
If it works for you all credit goes to @rtoijala in bug 876 as bisecting down to commit c352069b31b2dbb4d87503603d6c6d3ddd517df6 got me to the right place

**Comment by lectrode** [comment]
This bug affected me as well. In my testing, all ```execpi``` and ```texecpi``` functions were re-running their commands every interval, not just parsing the output. conky was also using 3.50% cpu every interval.

Commenting out ```conky::run_all_callbacks();``` at line 737 fixed the issue - the external commands are now called at their specified intervals. Conky is back to using ~0.38% cpu per interval.

**Comment by brndnmtthws** [comment]
This should be resolved with #1093.

**Comment by omexlu** [comment]
Hello,

I have installed conky over 'sudo apt-get install conky-all' (conky 1.10.8)
I have the same issue how can i fix that and where are the files located?
